### DISCUSSION TITLE
Download eQSL images

### DISCUSSION THREAD

**Discussion by R1BLH** [magicbug/Cloudlog#2166] [General]
The behaviour of the Download eQSL images tab is not entirely clear. Several questions have arisen:

1. Do I understand correctly that when downloading QSO data from the eQSL.cc website (Download QSOs tab) images are not downloaded and that they are only downloaded when viewing the log?

2. Is it possible to download all eQSL images? Clicking the buttons in the table more than 1000 times is tedious :)
![2023-05-25_15-01-10](https://github.com/magicbug/Cloudlog/assets/16667173/6a81dd4c-99c7-4209-96f2-c9e2352368c5)

3. When you click the button, two eQSLs are downloaded at once (the number of undownloaded eQSLs is reduced by 2, the table shows two rows less, see screenshot).

![2023-05-25_15-02-17](https://github.com/magicbug/Cloudlog/assets/16667173/0acba923-b5be-48df-8035-6cdb4173da20)

**Comment by phl0** [comment]
There should be a button at the bottom that allows for downloading of all outstanding eQSLs until timeout or rate limit. Are you on latest dev version?

**Reply by R1BLH** [reply]
> There should be a button at the bottom that allows for downloading of all outstanding eQSLs until timeout or rate limit. Are you on latest dev version?

There is no button.  I'll update it now.

You're absolutely right! Thank you!

**Reply by R1BLH** [reply]
I'm having a strange problem. Data upload and confirmation download work fine, but eQSL image download returns an error. Please have a look at the screenshots.
This is my second logbook for the same account.
![Снимок экрана от 2023-07-02 17-50-52](https://github.com/magicbug/Cloudlog/assets/16667173/41b681a2-0dbb-4cef-8fbf-ba7d16f57504)
![Снимок экрана от 2023-07-02 18-02-38](https://github.com/magicbug/Cloudlog/assets/16667173/d3152091-f3a5-453a-8c69-1e222018e2c7)

**Comment by MikleBas** [comment]
I do not have such tabs regarding QSL at all
Is it in the test assembly or is it in all?
If so, where can you find them?
Thank you

**Reply by phl0** [reply]
> I do not have such tabs regarding QSL at all Is it in the test assembly or is it in all? If so, where can you find them? Thank you

I guess you are on master branch?

**Reply by MikleBas** [reply]
Yes

**Reply by MikleBas** [reply]
![Screenshot (2)](https://github.com/magicbug/Cloudlog/assets/99086477/81a12ac5-0b9b-464f-acba-edef6232ecb5)

**Reply by phl0** [reply]
Then you will have to update your checkout now as there was a new release yesterday which merged the latest changes from dev to master branch. The buttons should be there now.

**Reply by MikleBas** [reply]
Hello!
yes, I updated, the menu appeared
There is one thing But - it falls off on a timeout, is it possible to give this command (or in the console) somehow in the cron?

**Reply by phl0** [reply]
Yeah that is due to the rate limit on eqsl.cc side. Nothing that we can change apparently. We habe to wait 10 seconds between images.

**Reply by magicbug** [reply]
Yep nothing we can do, you could script something so that it reruns on timeout but that's a server thing rather than a cloudlog PHP code thing

**Reply by MikleBas** [reply]
Hello
yes I understand
as an option - what if we apply a construction like
curl --silent https://<URL-To-Cloudlog>/index.php/eqsl/sync &>/dev/null (This is an example command, from the help file)
I would try, but I don’t understand where the button request goes, by which everything is downloaded at once

**Reply by AndreasK79** [reply]
@MikleBas the sync cron job only uploads QSOs and downloads confirmation. No cron job exists for downloading images.
